### DISCUSSION TITLE
"SMTP server details are not configured or are incorrect"

### DISCUSSION THREAD

**Discussion by torbengb** [lukevella/rallly#314] [Q&A]
I'm getting this error, "SMTP server details are not configured or are incorrect" and I can't figure out what's wrong.

Here's what I have done so far:

1. Linux host
2. running Docker
3. As [per documentation](https://github.com/lukevella/rallly#-quickstart-with-docker), I have cloned the git repo.
4. Edited `docker-compose.yml`: only changed port from `3000` to `8016` due to port in use already.
5. Edited `.env`:

```
NEXT_PUBLIC_BASE_URL=https://doodle.mydomain.tld:8016
#DATABASE_URL=postgres://your-database/db # this is commented out as described in https://github.com/lukevella/rallly#%EF%B8%8F-configuration
SECRET_PASSWORD=I-have-replaced-this-password
SUPPORT_EMAIL=myname@mydomain.tld
SMTP_HOST=smtp.google.com
# my domain is under Google Workspace, so I can normally use Gmail's SMTP here.
SMTP_PORT=587
SMTP_SECURE=true
SMTP_USER=myname@mydomain.tld
SMTP_PWD=custom app password from https://myaccount.google.com/apppasswords
```

6. `docker-compose up -d`
```
Creating rallly_rallly_db_1 ... done
Creating rallly_rallly_1    ... done
```

7. using Portainer to view the log:
```
yarn run v1.22.19
$ /usr/src/app/node_modules/.bin/prisma migrate deploy --schema prisma/schema.prisma
Prisma schema loaded from prisma/schema.prisma
Datasource "db": PostgreSQL database "db", schema "public" at "rallly_db:5432"
20 migrations found in prisma/migrations
No pending migrations to apply.
Done in 12.63s.
yarn run v1.22.19
$ next start
ready - started server on 0.0.0.0:3000, url: http://localhost:3000
SMTP server details are not configured or are incorrect
SMTP server details are not configured or are incorrect
```

Yes, that final line is present twice.

8. Kill + remove the container, repeat from step 3, same result again.

**Comment by seppal123** [comment]
I have the same issue , any solution to this so far ?

**Reply by pslestang** [reply]
Usually port 587 is for TLS so you should set option secure = false
If you really want smtp over ssl try to connect on port 465.

**Comment by lukevella** [comment]
I just pushed [a small change](https://github.com/lukevella/rallly/pull/316/files) that will log the full error coming from node-mailer. Hopefully, that will help identify the issue here. You will need to pull the latest commit from the main branch and rebuild the image.

**Reply by seppal123** [reply]
Thank you, this helped me to solve the problem.
Reason: Google wanted to auth via Web browser.
https://accounts.google.com/DisplayUnlockCaptcha on that local server did help and rallly can send now emails.

**Reply by torbengb** [reply]
> Reason: Google wanted to auth via Web browser. https://accounts.google.com/DisplayUnlockCaptcha on that local server

Of course that's it. Unfortunately, my local server is either a NAS box or a headless Linux machine, so from neither would I be able to start a local web browser. F*ck. I will need to use a different SMTP provider.
